ReactiveCocoa/Swift/SignalProducer.swift

```diff
@@ -370,6 +370,30 @@ extension SignalProducerType {
 	public func startWithInterrupted(interrupted: () -> ()) -> Disposable {
 		return start(Observer(interrupted: interrupted))
 	}
+	
+	/// Creates a new producer that emits an array that contains the latest N values that were emitted 
+	/// by the original producer as specified in 'capacity'. If intermediateResults is true the
+	/// the new producer will emit an array with each new value until it reaches capacity and begins 
+	/// removing the earliest value. If intermediateResults is false then the new producer will begin emitting
+	/// only when it has reached capacity.
+	@warn_unused_result(message="Did you forget to call `start` on the producer?")
+	public func latestValues(capacity:Int, intermediateResults: Bool = true) -> SignalProducer<[Value], Error> {
```

Comment:
Having a method behavior depend on a boolean parameter is a smell: it tells you that the method is actually doing 2 completely different things!
Instead, I would settle on one behavior, that allows users to get the other one by means of using another operator.